File: Puppet/Puppet/MSAnalyticsTranmissionTargetSelectorViewCell.m
#import "Constants.h"
#import "MSAnalyticsTranmissionTargetSelectorViewCell.h"

@interface MSAnalyticsTranmissionTargetSelectorViewCell ()

@property(weak, nonatomic) IBOutlet UISegmentedControl *transmissionTargetSelector;

@end

@implementation MSAnalyticsTranmissionTargetSelectorViewCell

- (void)awakeFromNib {
  [super awakeFromNib];
  _transmissionTargetMapping =
      @[ kMSEventPropertiesIdentifier, kMSTargetToken1, kMSTargetToken2, kMSRuntimeTargetToken ];
  _didSelectTransmissionTarget = ^() {
  };
  [_transmissionTargetSelector addTarget:self
                                  action:@selector(didSelectSegment)
                        forControlEvents:UIControlEventValueChanged];
}

- (NSString *)selectedTransmissionTarget {
  return self.transmissionTargetMapping[self.transmissionTargetSelector.selectedSegmentIndex];
}

- (void)didSelectSegment {
  self.didSelectTransmissionTarget();
}

@end

File: TestApps/Sasquatch/Sasquatch/ViewControllers/MSAnalyticsTranmissionTargetSelectorViewCell.swift
import UIKit

@objc(MSAnalyticsTranmissionTargetSelectorViewCell) class MSAnalyticsTranmissionTargetSelectorViewCell: UITableViewCell {

@IBOutlet weak var transmissionTargetSelector: UISegmentedControl!

  public var didSelectTransmissionTarget: (() -> Void)?
  public let eventPropertiesIdentifier = "Event Arguments"

  override func awakeFromNib() {
    super.awakeFromNib()
    didSelectTransmissionTarget = {_ in}
    transmissionTargetSelector.addTarget(self, action: #selector(onSegmentSelected), for: .valueChanged)
  }

  public func selectedTransmissionTarget() -> String! {
    return transmissionTargetSelector.titleForSegment(at: transmissionTargetSelector.selectedSegmentIndex)
  }

  public func transmissionTargets() -> [String]! {
    var targets = [String].init()
    for index in 0...transmissionTargetSelector.numberOfSegments - 1 {
      targets.append(transmissionTargetSelector.titleForSegment(at: index)!)
    }
    return targets
  }

  func onSegmentSelected() {
    didSelectTransmissionTarget?()
  }
}
